ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureClient.java

```diff
@@ -196,25 +239,152 @@ public void onResult(String consumerSessionId, Exception error) {
         });
     }
 
+    // endregion
+
+    // region Launch 3DS With App/Browser Switch Encapsulation
+
+//    /**
+//     * Verification is associated with a transaction amount and your merchant account. To specify a
+//     * different merchant account (or, in turn, currency), you will need to specify the merchant
+//     * account id when <a href="https://developers.braintreepayments.com/android/sdk/overview/generate-client-token">
+//     * generating a client token</a>
+//     * <p>
+//     * During lookup the original payment method nonce is consumed and a new one is returned,
+//     * which points to the original payment method, as well as the 3D Secure verification.
+//     * Transactions created with this nonce will be 3D Secure, and benefit from the appropriate
+//     * liability shift if authentication is successful or fail with a 3D Secure failure.
+//     *
+//     * The result of this verification will be returned to your {@link ThreeDSecureListener}.
+//     *
+//     * @param activity Android FragmentActivity
+//     * @param request  the {@link ThreeDSecureRequest} with information used for authentication.
+//     */
+//    public void performVerification(@NonNull final FragmentActivity activity, @NonNull final ThreeDSecureRequest request) {
```

Comment:
We updated this method to follow the listener result pattern, but realized that if the merchant invoked this method, and receives a result in their listener `onSuccess` method, they then nee to call `continuePerformVerification` from that listener `onSuccess` method, and would receive another result back in the same listener method. This seemed weird. Instead, we opted to leave the callback pattern for the `performVerification` method, and use the listener pattern for the methods that actually invoke a browser/app switch. This is documented in a code snipped in the `RESULT_API_MIGRATION.md`

Reply:
Left it commented out in case anyone thinks we should offer this with the listener instead. Otherwise will remove.

Reply:
I guess it depends if we want to add another `onLookupComplete` listener to the interface? We can be flexible, it doesn't have to exactly match the old way. There may be an opportunity here to enhance the 3DS developer experience.

Reply:
Yeah that might be nice for merchants who want to only use the listener pattern. I don't mind keeping listeners for browser/app switches and callbacks for API though. Also if we add that additional method to the listener interface, merchants using server-side 3DS would have to implement that method even though they don't need it. Can go either way - thoughts?

Reply:
qq: Do we need to uncomment this method?

Reply:
This method was added in case we wanted to follow the listener pattern for all methods. Since we are ok with callback only for non app/browser switch methods we can remove. I'll do that 👍